crates/common/src/raindex_client/vaults.rs

```diff
@@ -386,20 +393,23 @@ impl RaindexVault {
     /// const calldata = result.value;
     /// // Do something with the calldata
     /// ```
-    #[wasm_export(js_name = "getApprovalCalldata", unchecked_return_type = "Hex")]
+    #[wasm_export(
+        js_name = "getApprovalCalldata",
```

Comment:
probably for a follow-up but it seems useful to have a VaultDeposit type that could provide whatever calldatas needed, e.g.

```
const deposit = vault.createDeposit(amount);
const calldatas = deposit.getCalldatas(); // internally this would check the allowance and provide one or both calldatas
const approvalCalldata = calldatas?.approval;
const depositCalldata = calldatas.deposit;
```

Reply:
this seems like a good idea actually. i will create an issue for this